### DISCUSSION TITLE
How do you deal with 1.8k+ issues?

### DISCUSSION THREAD

**Discussion by htor** [zed-industries/zed#11320] [Feature Requests]
As of right now there are 1,821 open issues. That may not be much for some teams but for a small team like the Zed team I must imagine that's a lot to deal with. Or is it?

How do you deal with it? How do you prioritize tasks among this noise? What role does outside contributions have in the formation of the product? As a developer I'm curious!

**Comment by JosephTLyons** [comment]
To be super honest, it isn't easy. Every week, we have hundreds of issues opened, and I probably go through 500-1,000 notifications every week in just my GitHub inbox alone (not including Twitter, Emails, Discord, etc.). We do have some tools to help us parse all this feedback.

These scripts run and help us determine what the users want, based on upvotes, and we try to grab some items off these lists every so often, while working on the big items that we personally want.

- https://github.com/zed-industries/zed/issues/6953

We also have access to some AI tools that analyze our issues and in-app feedback that help distill it all into chunks of feedback that are more actionable and less time-consuming to get through.

Outside of that, it's a *lot* of manual work. @Moshyfawn and I do our best to triage all the issues, which includes adding labels, closing duplicates, providing technical support for issues that are more about the user not knowing that something exists, elevating things to the team that feel really important, like crashes, etc., but in the end, a lot of it can't be handled right now with a small team, so we do make a lot of gut-instinct decisions to let issues sit until the community upvotes them.

Ultimately, I'd love to have to do less manual triaging, but that would require us having more issue templates, and I wouldn't even know where to start with that - presumably the most popular issue types that keep surfacing, but our issues are mostly all wildly different things, with a few themes here and there.

Our system isn't perfect, it's pretty challenging getting through it all with an objective frame of mind, but we're doing our best. If anyone has any suggestions, coming from teams getting more issues, I'd love to get any tips or pointers - we are always trying to learn how to make this better. I think we could also use more community maintainers, but we'd probably need a better internal system for sharing knowledge before jumping on that - our current system is pretty informal at the moment.

**Reply by xpe** [reply]
I wonder if you might be able to close out as many as 50% of them by some kind of approach that does some combination of the following:

- "Hi, I'm an issue-triage bot. If this issue is still important to you, please let us know. Otherwise, we're going to close it so we can focus our efforts on the topics that we know are most important to the most people."
- "We really appreciate your feedback. Things be crazy right now. Here's our plan. We'd like _you_ to compare your issue with [these 100 top-ranked issues]. If you think your issue should be in that list, let us know. But if you think it is ok to wait and check back in a month, please add the `ok-to-wait` tag.

I just made these ideas up. They might be terrible. I have an inkling that some modification of them might help some teams.

Anyhow, I know the feeling of having a pile of issues. Think of it this way: they reflect (1) a sign of tremendous interest in Zed, (2) considerable care by people (enough to file an issue!), and (3) a healthy reminder of how much work is to be done.

**Comment by JosephTLyons** [comment]
Thanks for reaching out. We want to better support our community, and the best way for us to do that is through our official Discord. The team is active there and happy to help.

We’re closing the “Help and General Q&A” category on GitHub Discussions and directing all support questions to Discord going forward.

https://zed.dev/support
